### ISSUE TITLE
[BUG] No such file or directory: '/data/session AND InvalidSignature  AND cryptography.fernet.InvalidToken

### ISSUE THREAD

**Issue by bruvv** [benbusby/whoogle-search#597]
**Describe the bug**
```
ERROR:app:Exception on /element [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 113, in _verify_signature
    h.verify(data[-32:])
  File "/usr/local/lib/python3.8/site-packages/cryptography/hazmat/primitives/hmac.py", line 70, in verify
    ctx.verify(signature)
  File "/usr/local/lib/python3.8/site-packages/cryptography/hazmat/backends/openssl/hmac.py", line 76, in verify
    raise InvalidSignature("Signature did not match digest.")
cryptography.exceptions.InvalidSignature: Signature did not match digest.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/whoogle/app/routes.py", line 94, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 53, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 437, in element
    src_url = cipher_suite.decrypt(request.args.get('url').encode()).decode()
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 76, in decrypt
    return self._decrypt_data(data, timestamp, ttl, int(time.time()))
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 125, in _decrypt_data
    self._verify_signature(data)
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 115, in _verify_signature
    raise InvalidToken
cryptography.fernet.InvalidToken
ERROR:app:Exception on /element [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 113, in _verify_signature
    h.verify(data[-32:])
  File "/usr/local/lib/python3.8/site-packages/cryptography/hazmat/primitives/hmac.py", line 70, in verify
    ctx.verify(signature)
  File "/usr/local/lib/python3.8/site-packages/cryptography/hazmat/backends/openssl/hmac.py", line 76, in verify
    raise InvalidSignature("Signature did not match digest.")
cryptography.exceptions.InvalidSignature: Signature did not match digest.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/whoogle/app/routes.py", line 94, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 53, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 437, in element
    src_url = cipher_suite.decrypt(request.args.get('url').encode()).decode()
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 76, in decrypt
    return self._decrypt_data(data, timestamp, ttl, int(time.time()))
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 125, in _decrypt_data
    self._verify_signature(data)
  File "/usr/local/lib/python3.8/site-packages/cryptography/fernet.py", line 115, in _verify_signature
    raise InvalidToken
cryptography.fernet.InvalidToken
WARNING:root:Exception raised while handling cache file '/data/session/180230b0d272fdbaa6a91da05b969b2e'
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/cachelib/file.py", line 229, in set
    os.chmod(filename, self._mode)
FileNotFoundError: [Errno 2] No such file or directory: '/data/session/180230b0d272fdbaa6a91da05b969b2e'
```

**To Reproduce**
1. install latest tag using docker
2. use it and out of the blue it will shown an error

**Deployment Method**
- [ ] Heroku (one-click deploy)
- [X] Docker
- [ ] `run` executable
- [ ] pip/pipx
- [ ] Other: [describe setup]

**Version of Whoogle Search**
- [ ] Latest build from [source] (i.e. GitHub, Docker Hub, pip, etc)
- [X] Version 0.7.0 (latest from docker)
- [ ] Not sure

**Desktop (please complete the following information):**
 - OS: Windows 10, Mac, Android etc
 - Browser: Chrome, firefox, safari
 - Version: Different versions

**Comment by DUOLabs333** [comment]
Have you tried upgrading all of your dependencies?

**Comment by bruvv** [comment]
I'm running docker so no

**Comment by ngosang** [comment]
I have the same error Docker image `benbusby/whoogle-search:0.7.0`

**Comment by Martin555** [comment]
I get the same error with a manually installed instance. And I also tried installing with pip, which results in the same problem.
All dependencies are matched, all python packages are up to date and the time is correctly set. I also run other TLS encrypted websites on that server, which do not show cryptography issues.

```
$ pip3.9 -V
pip 22.0.3 from /home/martin/.local/lib/python3.9/site-packages/pip (python 3.9)

$ cat /etc/centos-release
CentOS Linux release 7.9.2009 (Core)
3.10.0-1160.53.1.el7.x86_64
```

**Comment by Martin555** [comment]
> Weird, I never saw this problem even once. Can you try pulling from git (or using `latest`)?

Sure, I just pulled from main branch and tried it again. It throwed the same `cryptography.fernet.InvalidToken` error. This time I also tried it with Python 3.8.11.

**Comment by DUOLabs333** [comment]
Can you try Python 3.9?

**Comment by DUOLabs333** [comment]
Yeah, I saw that.

**Comment by DUOLabs333** [comment]
I'm getting this problem as well --- I can't go on to the next page because I get a random string for my query.

**Comment by bruvv** [comment]
I'm not getting this issue anymore. Anyone else? Otherwise I'm closing this.

**Comment by ngosang** [comment]
It was fixed for me in 0.7.2
